scripts/zones/Valkurm_Dunes/npcs/qm2.lua

```diff
@@ -14,7 +14,10 @@ function onTrade(player,npc,trade)
 end;
 
 function onTrigger(player,npc)
-    if (not GetMobByID(MARCHELUTE):isSpawned() and player:getQuestStatus(SANDORIA,MESSENGER_FROM_BEYOND) == QUEST_ACCEPTED and not player:hasItem(1096)) then
+    local TOTD = VanadielTOTD();
+    local isNight = TOTD == dsp.time.NIGHT
+    
+    if (isNight and not GetMobByID(MARCHELUTE):isSpawned() and player:getQuestStatus(SANDORIA,MESSENGER_FROM_BEYOND) == QUEST_ACCEPTED and not player:hasItem(1096)) then
```

Comment:
```if VanadielTOTD() == dsp.time.NIGHT``` works and is just as intuitive to read. No need to make extra variables that are only used once.

Reply:
Sure, fixed.